---
title: "assignment"
output: html_document
date: "2023-04-24"
---
```{r}
suppressPackageStartupMessages({
  library(epiwraps)
  library(AnnotationHub)
  library(MotifDb)
  library(memes)
  library(universalmotif)
  library(ensembldb)
  library(ggplot2)
})
ah <- AnnotationHub(localHub=FALSE)
ensdb <- ah[["AH89211"]] # mouse ensembldb object
```

# Download the data
```{r, eval=FALSE}
options(timeout=1800)
download.file("https://ethz-ins.org/content/w7/atac.chr19.bam", "atac.chr19.bam", mode="wb")
download.file("https://ethz-ins.org/content/w7/atac.chr19.bam.bai", "atac.chr19.bam.bai",mode="wb")

download.file("https://ethz-ins.org/content/w7/w7_bw.zip", "w7_bw.zip")
unzip("w7_bw.zip")
```
#preparing tracks 
```{r}
bam <- "atac.chr19.bam"
# create a track using all fragments, computing the coverage at each position (and force Ensembl naming) binwidth for nucleotides
bam2bw(bam, output_bw = "full_cov.bw", paired=TRUE, binWidth=10L, forceSeqlevelsStyle = "Ensembl")
# create a track using all fragments, but computing the number of cuts/insertion sites at each position
epiwraps::bam2bw(bam, output_bw = "full_cuts.bw", paired=TRUE, binWidth=1L, type="ends", 
                 shift=c(4L,-5L), forceSeqlevelsStyle = "Ensembl")
# create a track using only nucleosome-free fragments, computing coverage
bam2bw(bam, output_bw = "NF_cov.bw", paired=TRUE, binWidth=5L, minFragLength=30, 
       maxFragLength=120, forceSeqlevelsStyle = "Ensembl")
# create a track using only nucleosome-free fragments, the number of cuts/insertion sites at each position
bam2bw(bam, output_bw = "NF_cuts.bw", paired=TRUE, binWidth=1L, type="ends", minFragLength=30, 
       maxFragLength=120, shift=c(4L,-5L), forceSeqlevelsStyle = "Ensembl")
# create a track using only mono-nucleosome fragments, computing coverage
bam2bw(bam, output_bw = "mono_cov.bw", paired=TRUE, binWidth=10L, minFragLength=140,
       maxFragLength=220, forceSeqlevelsStyle = "Ensembl")
# create a track using only mono-nucleosome fragments, computing centers
bam2bw(bam, output_bw = "mono_centers.bw", paired=TRUE, binWidth=5L, minFragLength=140,
       maxFragLength=220, type="center", forceSeqlevelsStyle = "Ensembl")
```

# Obtaining the sites- KLF4 motif
```{r}
#get KLF4 motif
motifs <- query(MotifDb, c("KLF4","Mus"))[[1]]
motif2 <- convert_motifs(motifs, class="TFBSTools-PFMatrix")
genome <- ah[["AH68356"]]

# get the sequence for chr19:
chr19 <- import(genome)["19"]

# find motif occurrences
moi <- motifmatchr::matchMotifs(motif2, chr19, out="positions", p.cutoff=1e-5)[[1]]
moi_2 <- motifmatchr::matchMotifs(motif2, chr19, out="positions", p.cutoff=1e-5)[[1]]

# convert to GRanges (when using motifmatchr)
moi <- as(setNames(moi,names(chr19)), "GRanges")
```
# Obtaining the sites- MAZ motif
```{r}
#get KLF4 motif
motifs_maz <- query(MotifDb, c("MAZ","Mus"))[[1]]
motif2_maz <- convert_motifs(motifs_maz, class="TFBSTools-PFMatrix")

# find motif occurrences
moi_maz <- motifmatchr::matchMotifs(motif2_maz, chr19, out="positions", p.cutoff=1e-5)[[1]]
moi_2_maz <- motifmatchr::matchMotifs(motif2_maz, chr19, out="positions", p.cutoff=1e-5)[[1]]

# convert to GRanges (when using motifmatchr)
moi_maz <- as(setNames(moi_maz,names(chr19)), "GRanges")
```


# Plotting signal around the motif occurence KLF4
```{r}
# prepare the list of tracks
tracks <- c("Full coverage"="full_cov.bw", "NF coverage"="NF_cov.bw", "All cuts"="full_cuts.bw",
            "NF cuts"="NF_cuts.bw", "Mono cov"="mono_cov.bw", "Mono centers"="mono_centers.bw")
# extract signals around the motif occurrences
sm <- signal2Matrix(tracks, moi, w=5, extend=500)

#plot the signals
plotEnrichedHeatmaps(sm, trim=0.95, colors = c("white", "darkred")) + ggtitle("KLF4 motif")

# renormalize using border backgrounds
smb <- renormalizeBorders(sm)

# plot the renormalized signals
plotEnrichedHeatmaps(smb, trim=0.95, colors = c("white", "darkred")) + ggtitle("KLF4 motif")
```
# Plotting signal around the motif occurence MAZ
```{r}
sm_maz <- signal2Matrix(tracks, moi_maz, w=5, extend=500)

#plot the signals
plotEnrichedHeatmaps(sm_maz, trim=0.95, colors = c("white", "darkred")) + ggtitle("MAZ motif")

# renormalize using border backgrounds
smb_maz <- renormalizeBorders(sm_maz)

# plot the renormalized signals
plotEnrichedHeatmaps(smb_maz, trim=0.95, colors = c("white", "darkred")) + ggtitle("MAZ motif")
```
